src/DataFrame/IO/Parquet/Thrift.hs

```diff
@@ -650,18 +654,18 @@ readLogicalType buf pos lastFieldId fieldStack = do
                     replicateM_ 2 (readField buf pos 0 [])
                     return ENUM_TYPE
                 5 -> do
-                    readDecimalType (DecimalType{decimalTypeScale = 0, decimalTypePrecision = 0}) buf pos 0 []
+                    readDecimalType 0 0 buf pos 0 []
```

Comment:
Explanation of changes in this function:
The problem with passing `LogicalType` to `readDecimalType` / `readTimeType` means that these functions have to deal with other types as well, even though they don't care about them.

I changed those helper functions to only accept the specific fields pertaining to the type they aim to process, which obviates the need of having something like `error "unsupported type"` in them.

Instead of passing `DecimalType` (which belongs to a wider sum type - so the `readDecimalType` function than has to deal with other constructors